fenix/app/src/main/java/org/mozilla/fenix/tabhistory/TabHistoryBottomSheetView.kt

```diff
@@ -0,0 +1,169 @@
+        ModalBottomSheetValue.Hidden,
+        skipHalfExpanded = true,
+        confirmValueChange = {
+            if (it == ModalBottomSheetValue.Hidden) {
+                onDismiss()
+            }
+            true
+        },
+    )
+
+    SideEffect {
+        scope.launch {
+            modalBottomSheetState.show()
+        }
+    }
+
+    TabHistoryBottomSheetLayout(
+        tabHistoryListItems,
+        selectedTabHistoryItem,
+        onClick,
+        modalBottomSheetState,
+        onDismiss,
+    )
+}
+
+@Composable
+@OptIn(ExperimentalMaterialApi::class)
+private fun TabHistoryBottomSheetLayout(
+    tabHistoryListItems: List<TabHistoryItem>,
+    selectedTabHistoryItem: TabHistoryItem? = null,
+    onClick: (TabHistoryItem) -> Unit,
+    sheetState: ModalBottomSheetState,
+    onDismiss: () -> Unit,
+) {
+    val listState = rememberLazyListState()
+    selectedTabHistoryItem?.let {
+        val coroutineScope = rememberCoroutineScope()
+        SideEffect {
+            coroutineScope.launch {
+                listState.scrollToItem(it.index)
+            }
+        }
+    }
+    ModalBottomSheetLayout(
+        modifier = Modifier.fillMaxSize(),
+        sheetState = sheetState,
+        sheetBackgroundColor = colorResource(id = R.color.fx_mobile_layer_color_1),
+        scrimColor = Color.Transparent,
+        sheetContent = {
+            FirefoxTheme {
+                Divider(
+                    color = Color.Gray,
+                    thickness = 3.dp,
+                    modifier = Modifier
+                        .padding(top = 10.dp, bottom = 10.dp)
+                        .align(Alignment.CenterHorizontally)
+                        .width(50.dp),
+                )
+                LazyColumn(
+                    state = listState,
+                ) {
+                    items(tabHistoryListItems) { item ->
+                        TabHistoryItemView(
+                            onClick = onClick,
+                            tabHistoryItem = item,
+                            selectedTabHistoryItem = selectedTabHistoryItem,
+                            sheetState = sheetState,
+                            onDismiss = onDismiss,
+                        )
+                    }
+                }
+            }
+        },
+    ) {
+    }
+}
+
+@Composable
+@OptIn(ExperimentalMaterialApi::class)
+private fun TabHistoryItemView(
+    onClick: (TabHistoryItem) -> Unit,
+    selectedTabHistoryItem: TabHistoryItem? = null,
+    tabHistoryItem: TabHistoryItem,
+    sheetState: ModalBottomSheetState,
+    onDismiss: () -> Unit,
+) {
+    val coroutineScope = rememberCoroutineScope()
+    AndroidView(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(
+                horizontal = dimensionResource(id = R.dimen.home_item_horizontal_margin),
+            ),
+        factory = { context ->
+            WidgetSiteItemView(context)
```

Comment:
Sorry for the drive-by: Perhaps we should consider using [`FaviconListItem`](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt#98).

Reply:
You shouldn't have to worry about reproducing the box around the favicon. We made changes in bookmarks and history to remove the box around the favicon and I am guessing we never updated the Tab History view.

Reply:
Done